src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -61,11 +61,12 @@ public function denormalize($data, $type, $format = null, array $context = [])
             throw NotNormalizableValueException::createForUnexpectedDataType('The data is neither an integer nor a string, you should pass an integer or a string that can be parsed as an enumeration case of type '.$type.'.', $data, [Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true);
         }
 
-        try {
-            return $type::from($data);
-        } catch (\ValueError $e) {
-            throw NotNormalizableValueException::createForUnexpectedDataType($e->getMessage(), $data, [Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true, $e->getCode(), $e);
+        $matched = $type::tryfrom($data);
+        if (!$matched) {
```

Comment:
```suggestion
        if (!$matched = $type::tryFrom($data)) {
```
Per our CS, assignments should be inlined in the if statements. Also makes use of the correct case for `tryFrom()`